### DISCUSSION TITLE
Citation style, German Society for Psychology

### DISCUSSION THREAD

**Discussion by majdgh92** [typst/typst#2391] [Q&A] [answered]
Hello,
i would like to use German Society for Psychology citation style (Deutsche Gesellschaft für Psychologie (DGPs))
it should be like this for books:
Myers, D. G. (2014). Psychologie (3. Aufl.). Berlin: Springer.
last name, abbreviated first name. (year). Book title (edition). Location: Publisher.
i tried to use .bib file with @ in the typst file but i can't get such a style.
Can someone please help. Thanks.

**Comment by xkevio** [comment]
When the next update arrives (0.9.0), you'll be able to use CSL (Citation Style Language) files to completely customize your bibliography.
Here's one such `.csl` file for your preferred style: https://github.com/citation-style-language/styles/blob/master/deutsche-gesellschaft-fur-psychologie.csl

**Reply by majdgh92** [reply]
Hello xkevio,
Thank you very much for your answer. The last time on 27.10 to give the paper to the professor.
I have strictly to use DGPs style. I think i have to use manual way until typst release the new method or is there a way with some type of script to do this temporarily?

**Reply by SekoiaTree** [reply]
~~You could compile the main branch yourself and use that now~~ apparently incorrect

**Reply by majdgh92** [reply]
I thank you very much for your solution.
I am using the online app from typst to write the papers.
could you send me the link to this branch. is it hard to used this compiled version ( from point of User Interface)?
Where can i find instructions of how to use this CSL in typst?

**Reply by PgBiel** [reply, marked as answer]
I believe compiling Typst manually won't work since the CSL work isn't done yet, that will probably take a few weeks still.

@majdgh92 I'd recommend doing this manually for now. You can use labels to be able to refer to things in your custom bibliography later, e.g.

```typ
- Stuff stuff, by John Smith <citation-1>
```

You can then use something like `#link(<citation-1>)[STUFF, stuff]` to refer/link to it later (...or earlier) in the document. (You can probably use functions to make this more seamless, or even (harder choice!) show rules on `ref` to make it so you can write just `@citation-1`.)

**Reply by majdgh92** [reply]
Thank you very much. I would do it manually ( about 10 to 16 times citation) until i write the next paper maybe after a couple of months.
I hope by that time typst released the new version.
